proxy/src/transport/tls/identity.rs

```diff
@@ -0,0 +1,73 @@
+use conduit_proxy_controller_grpc;
+use convert::TryFrom;
+use super::{DnsName, InvalidDnsName};
+
+/// An endpoint's identity.
+///
+/// `Clone` isn't derived because `Identity` should be `Arc`ed.
+#[derive(Debug, Eq, PartialEq)]
+pub struct Identity(DnsName);
+
+impl Identity {
+    /// Parses the given TLS identity, if provided.
+    ///
+    /// `controller_namespace` will be None if TLS is disabled.
+    ///
+    /// In the event of an error, the error is logged, so no detailed error
+    /// information is returned.
+    pub fn maybe_from(
```

Comment:
TIOLI: `try_from`?

Reply:
I didn't call this `try_from` because the return type is `Result<Option<T>, E>`, not `Result<T E>`.

Reply:
That makes sense, nevermind!